File: Task1_1.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    
    int N;
    cout << "Введите размер матрицы N: ";
    cin >> N;
    
    // Инициализация матрицы
    vector<vector<int>> matrix(N, vector<int>(N));
    for (int i = 0; i < N; ++i) {
        for (int j = 0; j < N; ++j) {
            matrix[i][j] = rand() % 51 - 10; // [-10, 40]
        }
    }
    
    // Вывод матрицы
    cout << "Исходная матрица:" << endl;
    for (const auto& row : matrix) {
        for (int num : row) {
            cout << num << "\t";
        }
        cout << endl;
    }
    
    // Поиск строк с 3+ отрицательными элементами
    vector<int> negativeRows;
    vector<int> negativeElements;
    vector<vector<int>> newMatrix;
    
    for (int i = 0; i < N; ++i) {
        int count = 0;
        vector<int> currentNegatives;
        
        for (int j = 0; j < N; ++j) {
            if (matrix[i][j] < 0) {
                count++;
                currentNegatives.push_back(matrix[i][j]);
            }
        }
        
        if (count >= 3) {
            negativeRows.push_back(i);
            negativeElements.insert(negativeElements.end(), currentNegatives.begin(), currentNegatives.end());
            newMatrix.push_back(matrix[i]);
        }
    }
    
    // Вывод результатов
    cout << "\nНомера строк с 3+ отрицательными элементами: ";
    for (int row : negativeRows) {
        cout << row << " ";
    }
    
    cout << "\nОтрицательные элементы из этих строк: ";
    for (int num : negativeElements) {
        cout << num << " ";
    }
    
    cout << "\n\nНовая матрица из найденных строк:" << endl;
    for (const auto& row : newMatrix) {
        for (int num : row) {
            cout << num << "\t";
        }
        cout << endl;
    }
    return 0;
}

File: Task1_1.rs
use rand::Rng;
use std::io;

fn main() {
    let mut rng = rand::thread_rng();
    
    println!("Введите размер матрицы N: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Ошибка чтения");
    let n: usize = input.trim().parse().expect("Ожидается число");
    
    // Инициализация матрицы
    let mut matrix = vec![vec![0; n]; n];
    for i in 0..n {
        for j in 0..n {
            matrix[i][j] = rng.gen_range(-10..=40);
        }
    }
    
    // Вывод матрицы
    println!("Исходная матрица:");
    for row in &matrix {
        for num in row {
            print!("{}\t", num);
        }
        println!();
    }
    
    // Поиск строк с 3+ отрицательными элементами
    let mut negative_rows = Vec::new();
    let mut negative_elements = Vec::new();
    let mut new_matrix = Vec::new();
    
    for (i, row) in matrix.iter().enumerate() {
        let mut count = 0;
        let mut current_negatives = Vec::new();
        
        for num in row {
            if *num < 0 {
                count += 1;
                current_negatives.push(*num);
            }
        }
        
        if count >= 3 {
            negative_rows.push(i);
            negative_elements.extend(current_negatives);
            new_matrix.push(row.clone());
        }
    }
    
    // Вывод результатов
    print!("\nНомера строк с 3+ отрицательными элементами: ");
    for row in &negative_rows {
        print!("{} ", row);
    }
    
    print!("\nОтрицательные элементы из этих строк: ");
    for num in &negative_elements {
        print!("{} ", num);
    }
    
    println!("\n\nНовая матрица из найденных строк:");
    for row in &new_matrix {
        for num in row {
            print!("{}\t", num);
        }
        println!();
    }
}
